docs: add admin MCP page (#5587)
mintlify/docs: ai/admin-mcp.mdx

---

> title: "Admin MCP"

Reviewer: Are we confirmed naming this `Admin MCP`?

> description: "Let AI tools edit your Mintlify documentation directly with the admin MCP server. Connect Claude, Cursor, or any MCP client to draft, save, and ship doc changes."

Reviewer: ```suggestion
description: "Let AI tools directly edit your content on Mintlify with the MCP server. Connect Claude, Cursor, or any MCP client to draft, save, and ship doc changes."
```

TBD on the confirmed name

> keywords: ["admin MCP", "write access", "AI editing", "Claude", "Cursor", "branch", "pull request"]

Reviewer: ```suggestion
keywords: ["MCP", "write access", "AI", "editing", "Claude", "Cursor", "branch", "pull request"]
```

---

## About the admin MCP

> The admin MCP server gives AI tools write access to your Mintlify documentation. Where the [documentation MCP server](/ai/model-context-protocol) lets AI tools read and search your published docs, the admin MCP lets them propose changes: edit pages, restructure navigation, update `docs.json`, and open pull requests.

Reviewer: ```suggestion
The admin MCP server gives AI tools write access to your Mintlify content. Where the [documentation MCP server](/ai/model-context-protocol) lets AI tools read and search your published content, the admin MCP lets them propose changes: edit pages, restructure navigation, update `docs.json`, and open pull requests.
```

Starting to replace `documentation` with `content` as part of expanding to cover more use cases.

> Connect any MCP client like Claude, Claude Code, or Cursor to the admin MCP and your assistant can collaborate with you on docs the same way it collaborates on code — except every change runs through a Mintlify branch and lands in your repository as a pull request.

Reviewer: ```suggestion
Connect any MCP client like Claude, Claude Code, or Cursor to the admin MCP to collaborate on your Mintlify content with the same tools you use to write code. When you use the MCP server, all changes happen on a branch and require a pull request to merge.
```

<Note>
  The admin MCP edits your documentation. Treat it like a developer with commit access. Connect it only from trusted AI tools and review every pull request before merging.
</Note>

### How it differs from the documentation MCP

|  | Documentation MCP | Admin MCP |
| :-- | :-- | :-- |
| **Audience** | Your end users | Your team |
| **Access** | Read and search published pages | Read, edit, restructure, save |
| **Endpoint** | `/mcp` on your docs domain | Hosted by Mintlify, scoped to your project |
| **Output** | Search results and page content | MDX edits, navigation changes, pull requests |

## Connect to the admin MCP

> Find your admin MCP URL on the [MCP server page](https://dashboard.mintlify.com/products/mcp) in your dashboard. Connecting requires an interactive OAuth login against your Mintlify account; the AI tool exchanges that login for a session token scoped to one project.

Reviewer: ```suggestion
You must have an interactive OAuth login against your Mintlify account to connect the admin MCP. Any AI tool that you connect exchanges your OAuth login for a session token scoped to one project.

Find your admin MCP URL on the [MCP server page](https://dashboard.mintlify.com/products/mcp) in your dashboard.
```

Putting the prerequisite first

<Tabs>
  <Tab title="Claude">
    <Steps>
      <Step title="Add the admin MCP as a custom connector">
        1. Navigate to the [Connectors](https://claude.ai/settings/connectors) page in the Claude settings.
        2. Select **Add custom connector**.
        3. Add your admin MCP server name and URL.
        4. Select **Add** and complete the OAuth login.
      </Step>
      <Step title="Use it in a chat">

>         Select the attachments button (the plus icon) and choose your admin MCP server. Claude can now call the admin MCP tools while answering your prompt.

Reviewer: ```suggestion
        1. Click the <Icon icon="plus" /> attachments button. 
        2. Select your admin MCP server.
        3. Prompt Claude to do a task that requires calling the admin MCP tools, like updating a page.
```

      </Step>
    </Steps>
  </Tab>
  <Tab title="Claude Code">
    Add the admin MCP server with the Claude Code CLI:

    ```bash
    claude mcp add --transport http mintlify-admin <admin-mcp-url>
    ```

    On first use, Claude Code opens a browser window to complete the OAuth login. After authenticating, the session is reused for subsequent calls.
  </Tab>
  <Tab title="Cursor">
    1. Open the command palette with <kbd>Command</kbd> \+ <kbd>Shift</kbd> \+ <kbd>P</kbd> (<kbd>Ctrl</kbd> \+ <kbd>Shift</kbd> \+ <kbd>P</kbd> on Windows).
    2. Search for **Open MCP settings** and select **Add custom MCP**.
    3. In `mcp.json`, add the admin MCP:

    ```json
    {
      "mcpServers": {
        "mintlify-admin": {
          "url": "<admin-mcp-url>"
        }
      }
    }
    ```

    4. Reload Cursor and complete the OAuth login when prompted.
  </Tab>
</Tabs>

## How a session works

> Every admin MCP session is bound to a single git branch. The flow is:

Reviewer: ```suggestion
Every admin MCP session is bound to a single Git branch. The flow is:
```

- **`search`** — Find lines matching a substring or regex across every page.
- **`edit_page`** — Apply a targeted edit to a page.
- **`write_page`** — Overwrite a page's full MDX content.

### Navigation

- **`list_nodes`** — Walk the navigation tree, filtered by type, language, version, or tab.
- **`create_node`** — Add a new page, group, tab, anchor, version, language, product, or dropdown.
- **`update_node`** — Update a node's properties in place (rename a group, change an icon, set a default version).
- **`move_node`** — Reparent or reorder a node, including renaming a page's path.
- **`delete_node`** — Remove a node from the navigation.

### Configuration

- **`update_config`** — Modify `docs.json` (theme, navigation roots, integrations, SEO settings).

### Session

- **`checkout`** — Bind the session to a branch.
- **`get_session_state`** — Inspect the current branch, edited files, and pending nav diff.
- **`diff`** — List all changes between the session and `main`.
- **`save`** — Open a pull request or commit to the session branch.
- **`discard_session`** — Drop the session and its in-flight changes.

## Example prompts

> Once the admin MCP is connected, you can drive it with natural-language prompts. For example:

Reviewer: ```suggestion
Once you connect the admin MCP, steer it with natural-language prompts. For example:
```

- _"Check out a branch called `add-billing-faq` and create a new page under the FAQ group titled 'Billing'. Draft answers for the five questions in this Linear issue: ..."_
- _"Find every page that mentions the deprecated `legacy_token` field and update the example to use `api_key` instead. Save as a PR titled 'docs: replace legacy\_token references'."_
- _"Reorganize the API reference: move the webhooks pages into a new group called 'Webhooks' and update the icons to match the rest of the section."_

## Best practices

<AccordionGroup>
  <Accordion title="Open the editor URL">
    Every `checkout` returns an `editorUrl`. Open it in a separate tab so you can watch the AI's changes render live in the dashboard editor while you prompt.
  </Accordion>

  <Accordion title="Review every PR">

>     The admin MCP is powerful enough to rewrite hundreds of pages in a single session. Before merging, read the PR diff and skim the rendered preview — don't rubber-stamp large changes.

Reviewer: ```suggestion
    The admin MCP is can make large edits in a single session like rewriting hundreds of pages. Before merging, read the PR diff and review the rendered preview. Don't merge changes without a human review.
```

  </Accordion>

  <Accordion title="Use slugs for branch names">
    Pass a `slug` to `checkout` (e.g. `add-quickstart`) so the auto-generated branch is human-readable. Without it, the branch name is derived from the session token and is hard to recognize in your repo.
  </Accordion>

  <Accordion title="Keep sessions focused">
    One session, one logical change. Smaller sessions produce smaller, easier-to-review pull requests. Use `discard_session` and `checkout` again to pivot to unrelated work.
  </Accordion>
</AccordionGroup>

<Warning>

>   Sessions hold an in-memory branch on the Mintlify side. If a session is abandoned without a `save` or `discard_session`, the branch persists until your next checkout overwrites it. Avoid leaving stale `admin-mcp/*` branches in your repository — clean them up periodically.

Reviewer: ```suggestion
  Sessions hold an in-memory branch on the Mintlify side. If you abandon a session without saving or discarding it, the branch persists until your next checkout overwrites it. Avoid leaving stale `admin-mcp/*` branches in your repository. Clean them up periodically.
```

> </Warning>

Reviewer: ```suggestion
</Note>
```
